### ISSUE TITLE
 Warning message: In quit... system call failed: Cannot allocate memory

### ISSUE THREAD

**Issue by ctlamb** [Azure/doAzureParallel#322]
Some of my nodes are failing with this error:

Warning message:
In quit(save = "yes", status = workerErrorStatus, runLast = FALSE) :
  system call failed: Cannot allocate memory

Does this mean I need a CPU with more memory?

**Comment by brnleehng** [comment]
Hi @ctlamb 

Yes, this means you will need a CPU with more memory. I suggest measuring the memory usage for each task so you have a benchmark of what Azure VM to use.

Thanks
Brian

**Comment by ctlamb** [comment]
Excellent, will do. In the meantime, I tried to use a machine with slightly more memory ("vmSize" = "Standard_E4_v3"),, but I am running into the following error after I run foreach (this error doesn't occur with "vmSize" = "Standard_DS12_v2")

`##Error: No automatic parser available for 7b/.`

**Comment by brnleehng** [comment]
What region are you in? It could be possible that Standard_E4_v3 is not available in your region. 
Is this happening during makeCluster?

**Comment by ctlamb** [comment]
I'm in West US. The error throws in `foreach`

It looks like my tasks only use a max of 8GB of RAM, so the 28GB of ram I had in the "Standard_DS12_v2" should've been plenty. hmmmm. Not sure what's going on here

**Comment by ctlamb** [comment]
Is there a better/preferred package I could use to measure the memory usage?

**Comment by ctlamb** [comment]
Now getting `Error: No automatic parser available for 7b/.` even when I use the D12 machine now. ugghh, always hard to trouble shoot one issue (memory) when another pops up. Any thoughts? I could start a new thread if its easier

**Comment by brnleehng** [comment]
I don't have a preferred package for measuring memory usage. 
Where exactly is this error occuring? Is this when the foreach is getting results?

If you have a cluster configuration file and a reproducible sample, I will work on identifying the issue

**Comment by simon-tarr** [comment]
This is the same as issue #315.
I've spent many an hour pulling my hair out over this issue and I've no idea what's causing the issue. I've provided a lot of qualitative information in #315 but haven't had time to build a fully reproducible example at the scale which I think is generating the error.

@ctlamb is your workflow using resource files uploaded to Azure storage? My workflow is and I haven't been able to determine whether the 7b error still occurs when not using resource files. I'd like to attempt to rule out whether resource files could be contributing in some way.

**Comment by ctlamb** [comment]
Yes I am uploading and downloading data to Azure storage in my workflow. I do wonder if this was an internet issue? My internet speed was recently upgraded, and I haven't got the 7b error since..but thats only based on 5-10 different tries so far. Will update if anything changes

**Comment by simon-tarr** [comment]
> Yes I am uploading and downloading data to Azure storage in my workflow. I do wonder if this was an internet issue? My internet speed was recently upgraded, and I haven't got the 7b error since..but thats only based on 5-10 different tries so far. Will update if anything changes

Thanks for the extra information. My latest post at #315 documents the return of the dreaded 7b error.

I considered your idea here as well. However, my university network is a gigabit connection and it's rock stable. My home internet is a 100Mb fire connection which is also super reliable (for the most part).

I wonder if there's a limit to the number of connections Batch/HTTR can accept from a single IP address? I'm currently running 2 pools on my laptop (home network) and three on my uni workstation all day and they've been stable all day. If I try and run any more pools than this on either machine, the 7b error will return almost instantly. It's very strange...

**Comment by brnleehng** [comment]
Are all of your workflows in interactive mode? (Waiting for the job to be done)

Thanks,
Brian

**Comment by simon-tarr** [comment]
Mine is, yes.

**Comment by simon-tarr** [comment]
Any news on the status of this error? It's still happening to me with frustrating regularity.

Thanks!
